test/infrastructure/docker/internal/controllers/dockermachine_controller.go

```diff
@@ -231,7 +231,9 @@ func (r *DockerMachineReconciler) reconcileNormal(ctx context.Context, cluster *
 
 	// Create the machine if not existing yet
 	if !externalMachine.Exists() {
-		if err := externalMachine.Create(ctx, role, machine.Spec.Version, dockerMachine.Spec.ExtraMounts); err != nil {
+		// NOTE: FailureDomains don't mean much in CAPD since it's all local, but we are setting a label on
+		// each container, so we can check placement.
+		if err := externalMachine.Create(ctx, dockerMachine.Spec.CustomImage, role, machine.Spec.Version, docker.FailureDomainLabel(machine.Spec.FailureDomain), dockerMachine.Spec.ExtraMounts); err != nil {
```

Comment:
Does this also work when FailureDomain is nil? (not sure if we still have a test for it)

Reply:
Yes, it works (it returns no labels)

Reply:
Not sure what I thought it's very obvious in the `FailureDomainLabel` func